src/compiler/typescript_to_rust/azle_generate/src/canister_method/errors/mismatched_post_deploy_params.rs

```diff
@@ -0,0 +1,220 @@
+
+        let line_number = source_map.get_line_number(first_span);
+        let origin = source_map.get_origin(first_span);
+        let offset = total_range.0.to_usize() - source_map.get_range(first_span).0;
+
+        let ranges = get_ranges(ordered_fn_decls, offset);
+
+        Self {
+            origin,
+            line_number,
+            source,
+            ranges,
+        }
+    }
+
+    pub fn to_string(&self) -> String {
+        let error_snippet = Snippet {
+            title: Some(Annotation {
+                label: Some("params for $init and $postUpgrade must be exactly the same"),
+                id: None,
+                annotation_type: AnnotationType::Error,
+            }),
+            slices: vec![Slice {
+                source: &self.source,
+                line_start: self.line_number,
+                origin: Some(&self.origin),
+                fold: true,
+                annotations: vec![
+                    SourceAnnotation {
+                        label: "these params",
+                        annotation_type: AnnotationType::Error,
+                        range: self.ranges[0],
+                    },
+                    SourceAnnotation {
+                        label: "and these params do not match",
+                        annotation_type: AnnotationType::Error,
+                        range: self.ranges[1],
+                    },
+                ],
+            }],
+            footer: vec![Annotation {
+                label: Some(
+                    "update the params for either $init or $postUpgrade to match the other",
+                ),
+                id: None,
+                annotation_type: AnnotationType::Help,
+            }],
+            opt: FormatOptions {
+                color: true,
+                ..Default::default()
+            },
+        };
+
+        format!("{}", DisplayList::from(error_snippet))
+    }
+}
+
+impl std::error::Error for MismatchedPostDeployParams {}
+
+impl std::fmt::Display for MismatchedPostDeployParams {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.to_string())
+    }
+}
+
+impl From<MismatchedPostDeployParams> for crate::Error {
+    fn from(error: MismatchedPostDeployParams) -> Self {
+        Self::MismatchedPostDeployParams(error)
+    }
+}
+
+impl From<MismatchedPostDeployParams> for Vec<crate::Error> {
+    fn from(error: MismatchedPostDeployParams) -> Self {
+        vec![crate::Error::MismatchedPostDeployParams(error)]
+    }
+}
+
+fn order_fn_decls<'a>(
+    a: &'a SmAnFnDecl<'a>,
+    b: &'a SmAnFnDecl<'a>,
+) -> (&'a SmAnFnDecl<'a>, &'a SmAnFnDecl<'a>) {
+    let a_start_byte_pos = get_start_byte_pos(a);
+    let b_start_byte_pos = get_start_byte_pos(b);
+
+    if a_start_byte_pos.lt(&b_start_byte_pos) {
+        (a, b)
+    } else {
+        (b, a)
+    }
+}
+
+fn get_start_byte_pos(sm_an_fn_decl: &SmAnFnDecl) -> BytePos {
```

Comment:
Get rid of the snippets, or comment them out. Let's just keep it simple

Reply:
Done in c1aea753d74f473388ac09c48f1a244421af856a